frontend/src/metabase/query_builder/components/dataref/DatabaseTablesPane/DatabaseTablesPane.tsx

```diff
@@ -0,0 +1,107 @@
+import { ngettext, msgid } from "ttag";
+import _ from "underscore";
+
+import Tables from "metabase/entities/tables";
+import Questions from "metabase/entities/questions";
+import type { Card } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+import Database from "metabase-lib/lib/metadata/Database";
+import {
+  NodeListItemLink,
+  NodeListItemName,
+  NodeListItemIcon,
+  NodeListTitle,
+  NodeListContainer,
+  NodeListIcon,
+  NodeListTitleText,
+} from "../NodeList.styled";
+import { ModelId } from "./DatabaseTablesPane.styled";
+
+interface DatabaseTablesPaneProps {
+  show: (type: string, item: unknown) => void;
+  database: Database;
+  models: Card[];
+}
+
+const DatabaseTablesPane = ({
+  database,
+  show,
+  models,
+}: DatabaseTablesPaneProps) => {
+  const tables = useMemo(
+    () => database.tables.sort((a, b) => a.name.localeCompare(b.name)),
+    [database.tables],
+  );
+  models = useMemo(
+    () => models?.sort((a, b) => a.name.localeCompare(b.name)),
+    [models],
+  );
+  return models ? (
+    <NodeListContainer>
+      {models?.length ? (
+        <>
+          <NodeListTitle>
+            <NodeListIcon name="model" />
+            <NodeListTitleText>
+              {ngettext(
+                msgid`${models.length} model`,
+                `${models.length} models`,
+                models.length,
+              )}
+            </NodeListTitleText>
+          </NodeListTitle>
+          <ul>
+            {models.map(model => (
+              <li key={model.id}>
+                <NodeListItemLink onClick={() => show("model", model)}>
+                  <NodeListItemIcon name="model" />
+                  <NodeListItemName>{model.name}</NodeListItemName>
+                  <ModelId>{`#${model.id}`}</ModelId>
+                </NodeListItemLink>
+              </li>
+            ))}
+          </ul>
+          <br></br>
+        </>
+      ) : null}
+      <NodeListTitle>
+        <NodeListIcon name="table" />
+        <NodeListTitleText>
+          {ngettext(
+            msgid`${tables.length} table`,
+            `${tables.length} tables`,
+            tables.length,
+          )}
+        </NodeListTitleText>
+      </NodeListTitle>
+      <ul>
+        {tables.map(table => (
+          <li key={table.id}>
+            <NodeListItemLink onClick={() => show("table", table)}>
+              <NodeListItemIcon name="table" />
+              <NodeListItemName>{table.name}</NodeListItemName>
+            </NodeListItemLink>
+          </li>
+        ))}
+      </ul>
+    </NodeListContainer>
+  ) : null;
+};
+
+export default _.compose(
+  Tables.loadList({
+    query: (_state: State, props: DatabaseTablesPaneProps) => ({
+      dbId: props.database?.id,
+    }),
+    loadingAndErrorWrapper: false,
+  }),
+  Questions.loadList({
+    query: (_state: State, props: DatabaseTablesPaneProps) => ({
+      model: true,
+      dbId: props.database?.id, // TODO: why could this be undefined?
```

Comment:
Should this `TODO` be removed? Either we expect `database` to be `undefined`/`null` or we fix the cause to not make it happen.

Reply:
Done. It's not logically possible to reach this pane without first loading a database.
https://github.com/metabase/metabase/pull/25494/commits/95cb8194f7133b669b9c7e6c5a34d61a0e35b6d9